### ISSUE TITLE
False positive for inherited descriptive annotations

### ISSUE THREAD

**Issue by marcospassos** [pepakriz/phpstan-exception-rules#67]
Inherited description annotations are incorrectly reported as unused:

```php
interface ClassInterface {
   /*
    * @throws \LogicException Description.
    */
    public function foo()
}

class ChildClass implements ClassInterface {
    public function foo(){
        throw new \LogicException(); // Unused @throws LogicException
    } 
}
```

**Comment by pepakriz** [comment]
This is expected behavior since phpstan v0.11.0 (PR https://github.com/phpstan/phpstan/pull/1533). If you omit phpdoc, it is inherited from the parent method. So your code is interpreted in the same way like:

```php
interface ClassInterface {
   /*
    * @throws \RuntimeException Description.
    */
    public function foo();
}

class ChildClass implements ClassInterface {
   /*
    * @throws \RuntimeException Description.
    */
    public function foo(){} // Unused @throws RuntimeException
}
```

**Comment by marcospassos** [comment]
My example was flawed, I've updated it. If you throw an exception declared in the parent method, it should not warn about that exception in the concrete method if it is descriptive.

**Comment by marcospassos** [comment]
Pending https://github.com/phpstan/phpstan/pull/1980

**Comment by marcospassos** [comment]
Fixed by #94.
